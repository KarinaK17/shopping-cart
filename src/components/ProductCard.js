import React from "react";
import cart from "../images/cart.svg";

const ProductCard = (props) => {
  if (props.quantity === 0) {
    return (
      <div className="Product-card">
        <div className="image-container">
          <img className="product-img" src={props.src} alt={props.name}></img>
          <p className="product-description">{props.description}</p>
        </div>
        <p>Price: {props.price} $</p>
        <div>
          <img
            data-testid={`cart-${props.index}`}
            className="cart"
            src={cart}
            alt="shopping cart"
            onClick={() => {
              props.handleCartClick(props.name);
            }}
          ></img>
        </div>
      </div>
    );
  } else {
    return (
      <div className="Product-card">
        <div className="image-container">
          <img className="product-img" src={props.src} alt={props.name}></img>
          <p className="product-description">{props.description}</p>
        </div>
        <p>Price: {props.price} $</p>
        <div className="input-container">
          <button
            onClick={() => {
              props.handleDecrement(props.name);
            }}
          >
            â€“
          </button>
          <input
            data-testid={`input-${props.index}`}
            type="tel"
            value={props.quantity}
            name="quantity"
            pattern={/^\d+$/}
            onChange={(e) => {
              props.handleQuantityChange(e, props.name);
            }}
          ></input>
          <button
            onClick={() => {
              props.handleIncrement(props.name);
            }}
          >
            +
          </button>
        </div>
      </div>
    );
  }
};

export default ProductCard;
