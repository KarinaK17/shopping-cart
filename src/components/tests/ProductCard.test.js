import { render } from "@testing-library/react";
import React from "react";
import ProductCard from "../ProductCard";

describe("ProductCard component", () => {
  it("renders correctly (if quantity is 0, card renders with cart)", () => {
    const { container } = render(<ProductCard quantity={0} />);
    expect(container).toMatchSnapshot();
  });

  it("renders correctly (if quantity is bigger than 0, card renders with input)", () => {
    const { container } = render(<ProductCard quantity={1} />);
    expect(container).toMatchSnapshot();
  });
});
