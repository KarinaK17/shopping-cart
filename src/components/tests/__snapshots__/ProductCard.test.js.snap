// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard component renders correctly (if quantity is 0, card renders with cart) 1`] = `
<div>
  <div
    class="Product-card"
  >
    <div
      class="image-container"
    >
      <img
        class="product-img"
      />
      <p
        class="product-description"
      />
    </div>
    <p>
      Price: 
       $
    </p>
    <div>
      <img
        alt="shopping cart"
        class="cart"
        data-testid="cart-undefined"
        src="cart.svg"
      />
    </div>
  </div>
</div>
`;

exports[`ProductCard component renders correctly (if quantity is bigger than 0, card renders with input) 1`] = `
<div>
  <div
    class="Product-card"
  >
    <div
      class="image-container"
    >
      <img
        class="product-img"
      />
      <p
        class="product-description"
      />
    </div>
    <p>
      Price: 
       $
    </p>
    <div
      class="input-container"
    >
      <button>
        â€“
      </button>
      <input
        data-testid="input-undefined"
        name="quantity"
        pattern="/^\\\\d+$/"
        type="tel"
        value="1"
      />
      <button>
        +
      </button>
    </div>
  </div>
</div>
`;
